File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-option-shared-5-4/CreditEventNotice.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CreditEventNotice : ISerialized
    {
        public CreditEventNotice(XmlNode xmlNode)
        {
            XmlNode notifyingPartyNode = xmlNode.SelectSingleNode("notifyingParty");
            
            if (notifyingPartyNode != null)
            {
                if (notifyingPartyNode.Attributes["href"] != null || notifyingPartyNode.Attributes["id"] != null) 
                {
                    if (notifyingPartyNode.Attributes["id"] != null) 
                    {
                        notifyingPartyIDRef_ = notifyingPartyNode.Attributes["id"].Value;
                        NotifyingParty ob = new NotifyingParty(notifyingPartyNode);
                        IDManager.SetID(notifyingPartyIDRef_, ob);
                    }
                    else if (notifyingPartyNode.Attributes["href"] != null)
                    {
                        notifyingPartyIDRef_ = notifyingPartyNode.Attributes["href"].Value;
                    }
                    else
                    {
                        notifyingParty_ = new NotifyingParty(notifyingPartyNode);
                    }
                }
                else
                {
                    notifyingParty_ = new NotifyingParty(notifyingPartyNode);
                }
            }
            
        
            XmlNode businessCenterNode = xmlNode.SelectSingleNode("businessCenter");
            
            if (businessCenterNode != null)
            {
                if (businessCenterNode.Attributes["href"] != null || businessCenterNode.Attributes["id"] != null) 
                {
                    if (businessCenterNode.Attributes["id"] != null) 
                    {
                        businessCenterIDRef_ = businessCenterNode.Attributes["id"].Value;
                        BusinessCenter ob = new BusinessCenter(businessCenterNode);
                        IDManager.SetID(businessCenterIDRef_, ob);
                    }
                    else if (businessCenterNode.Attributes["href"] != null)
                    {
                        businessCenterIDRef_ = businessCenterNode.Attributes["href"].Value;
                    }
                    else
                    {
                        businessCenter_ = new BusinessCenter(businessCenterNode);
                    }
                }
                else
                {
                    businessCenter_ = new BusinessCenter(businessCenterNode);
                }
            }
            
        
            XmlNode publiclyAvailableInformationNode = xmlNode.SelectSingleNode("publiclyAvailableInformation");
            
            if (publiclyAvailableInformationNode != null)
            {
                if (publiclyAvailableInformationNode.Attributes["href"] != null || publiclyAvailableInformationNode.Attributes["id"] != null) 
                {
                    if (publiclyAvailableInformationNode.Attributes["id"] != null) 
                    {
                        publiclyAvailableInformationIDRef_ = publiclyAvailableInformationNode.Attributes["id"].Value;
                        PubliclyAvailableInformation ob = new PubliclyAvailableInformation(publiclyAvailableInformationNode);
                        IDManager.SetID(publiclyAvailableInformationIDRef_, ob);
                    }
                    else if (publiclyAvailableInformationNode.Attributes["href"] != null)
                    {
                        publiclyAvailableInformationIDRef_ = publiclyAvailableInformationNode.Attributes["href"].Value;
                    }
                    else
                    {
                        publiclyAvailableInformation_ = new PubliclyAvailableInformation(publiclyAvailableInformationNode);
                    }
                }
                else
                {
                    publiclyAvailableInformation_ = new PubliclyAvailableInformation(publiclyAvailableInformationNode);
                }
            }
            
        
        }
        
    
        #region NotifyingParty_
        private NotifyingParty notifyingParty_;
        public NotifyingParty NotifyingParty_
        {
            get
            {
                if (this.notifyingParty_ != null)
                {
                    return this.notifyingParty_; 
                }
                else if (this.notifyingPartyIDRef_ != null)
                {
                    notifyingParty_ = IDManager.getID(notifyingPartyIDRef_) as NotifyingParty;
                    return this.notifyingParty_; 
                }
                else
                {
                      return this.notifyingParty_; 
                }
            }
            set
            {
                if (this.notifyingParty_ != value)
                {
                    this.notifyingParty_ = value;
                }
            }
        }
        #endregion
        
        public string notifyingPartyIDRef_ { get; set; }
        #region BusinessCenter_
        private BusinessCenter businessCenter_;
        public BusinessCenter BusinessCenter_
        {
            get
            {
                if (this.businessCenter_ != null)
                {
                    return this.businessCenter_; 
                }
                else if (this.businessCenterIDRef_ != null)
                {
                    businessCenter_ = IDManager.getID(businessCenterIDRef_) as BusinessCenter;
                    return this.businessCenter_; 
                }
                else
                {
                      return this.businessCenter_; 
                }
            }
            set
            {
                if (this.businessCenter_ != value)
                {
                    this.businessCenter_ = value;
                }
            }
        }
        #endregion
        
        public string businessCenterIDRef_ { get; set; }
        #region PubliclyAvailableInformation_
        private PubliclyAvailableInformation publiclyAvailableInformation_;
        public PubliclyAvailableInformation PubliclyAvailableInformation_
        {
            get
            {
                if (this.publiclyAvailableInformation_ != null)
                {
                    return this.publiclyAvailableInformation_; 
                }
                else if (this.publiclyAvailableInformationIDRef_ != null)
                {
                    publiclyAvailableInformation_ = IDManager.getID(publiclyAvailableInformationIDRef_) as PubliclyAvailableInformation;
                    return this.publiclyAvailableInformation_; 
                }
                else
                {
                      return this.publiclyAvailableInformation_; 
                }
            }
            set
            {
                if (this.publiclyAvailableInformation_ != value)
                {
                    this.publiclyAvailableInformation_ = value;
                }
            }
        }
        #endregion
        
        public string publiclyAvailableInformationIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-option-shared-5-4/CreditEventNotice.cpp
// CreditEventNotice.cpp 
#include "CreditEventNotice.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CreditEventNotice::CreditEventNotice(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //notifyingPartyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* notifyingPartyNode = xmlNode->FirstChildElement("notifyingParty");

   if(notifyingPartyNode){notifyingPartyIsNull_ = false;}
   else{notifyingPartyIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- notifyingPartyNode , address : " << notifyingPartyNode << std::endl;
   #endif
   if(notifyingPartyNode)
   {
      if(notifyingPartyNode->Attribute("href") || notifyingPartyNode->Attribute("id"))
      {
          if(notifyingPartyNode->Attribute("id"))
          {
             notifyingPartyIDRef_ = notifyingPartyNode->Attribute("id");
             notifyingParty_ = boost::shared_ptr<NotifyingParty>(new NotifyingParty(notifyingPartyNode));
             notifyingParty_->setID(notifyingPartyIDRef_);
             IDManager::instance().setID(notifyingPartyIDRef_,notifyingParty_);
          }
          else if(notifyingPartyNode->Attribute("href")) { notifyingPartyIDRef_ = notifyingPartyNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { notifyingParty_ = boost::shared_ptr<NotifyingParty>(new NotifyingParty(notifyingPartyNode));}
   }

   //businessCenterNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* businessCenterNode = xmlNode->FirstChildElement("businessCenter");

   if(businessCenterNode){businessCenterIsNull_ = false;}
   else{businessCenterIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- businessCenterNode , address : " << businessCenterNode << std::endl;
   #endif
   if(businessCenterNode)
   {
      if(businessCenterNode->Attribute("href") || businessCenterNode->Attribute("id"))
      {
          if(businessCenterNode->Attribute("id"))
          {
             businessCenterIDRef_ = businessCenterNode->Attribute("id");
             businessCenter_ = boost::shared_ptr<BusinessCenter>(new BusinessCenter(businessCenterNode));
             businessCenter_->setID(businessCenterIDRef_);
             IDManager::instance().setID(businessCenterIDRef_,businessCenter_);
          }
          else if(businessCenterNode->Attribute("href")) { businessCenterIDRef_ = businessCenterNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { businessCenter_ = boost::shared_ptr<BusinessCenter>(new BusinessCenter(businessCenterNode));}
   }

   //publiclyAvailableInformationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* publiclyAvailableInformationNode = xmlNode->FirstChildElement("publiclyAvailableInformation");

   if(publiclyAvailableInformationNode){publiclyAvailableInformationIsNull_ = false;}
   else{publiclyAvailableInformationIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- publiclyAvailableInformationNode , address : " << publiclyAvailableInformationNode << std::endl;
   #endif
   if(publiclyAvailableInformationNode)
   {
      if(publiclyAvailableInformationNode->Attribute("href") || publiclyAvailableInformationNode->Attribute("id"))
      {
          if(publiclyAvailableInformationNode->Attribute("id"))
          {
             publiclyAvailableInformationIDRef_ = publiclyAvailableInformationNode->Attribute("id");
             publiclyAvailableInformation_ = boost::shared_ptr<PubliclyAvailableInformation>(new PubliclyAvailableInformation(publiclyAvailableInformationNode));
             publiclyAvailableInformation_->setID(publiclyAvailableInformationIDRef_);
             IDManager::instance().setID(publiclyAvailableInformationIDRef_,publiclyAvailableInformation_);
          }
          else if(publiclyAvailableInformationNode->Attribute("href")) { publiclyAvailableInformationIDRef_ = publiclyAvailableInformationNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { publiclyAvailableInformation_ = boost::shared_ptr<PubliclyAvailableInformation>(new PubliclyAvailableInformation(publiclyAvailableInformationNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<NotifyingParty> CreditEventNotice::getNotifyingParty()
{
   if(!this->notifyingPartyIsNull_){
        if(notifyingPartyIDRef_ != ""){
             return boost::shared_static_cast<NotifyingParty>(IDManager::instance().getID(notifyingPartyIDRef_));
        }else{
             return this->notifyingParty_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NotifyingParty>();
   }
}
boost::shared_ptr<BusinessCenter> CreditEventNotice::getBusinessCenter()
{
   if(!this->businessCenterIsNull_){
        if(businessCenterIDRef_ != ""){
             return boost::shared_static_cast<BusinessCenter>(IDManager::instance().getID(businessCenterIDRef_));
        }else{
             return this->businessCenter_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<BusinessCenter>();
   }
}
boost::shared_ptr<PubliclyAvailableInformation> CreditEventNotice::getPubliclyAvailableInformation()
{
   if(!this->publiclyAvailableInformationIsNull_){
        if(publiclyAvailableInformationIDRef_ != ""){
             return boost::shared_static_cast<PubliclyAvailableInformation>(IDManager::instance().getID(publiclyAvailableInformationIDRef_));
        }else{
             return this->publiclyAvailableInformation_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PubliclyAvailableInformation>();
   }
}
}

